src/material-experimental/mdc-checkbox/checkbox.scss

```diff
@@ -0,0 +1,42 @@
+@import '@material/checkbox/functions';
+@import '@material/checkbox/mixins';
+@import '@material/form-field/mixins';
+@import '@material/ripple/variables';
+@import '../mdc-helpers/mdc-helpers';
+
+@include mdc-checkbox-without-ripple($query: $mat-base-styles-query);
+@include mdc-form-field-core-styles($query: $mat-base-styles-query);
+
+.mat-mdc-checkbox {
+  // stylelint-disable selector-class-pattern
+
+  .mdc-checkbox:hover
+      .mdc-checkbox__native-control:not([disabled]) ~ .mdc-checkbox__background::before {
+    opacity: map-get($mdc-ripple-dark-ink-opacities, hover);
+    transform: scale(2.75, 2.75);
+    transition: mdc-checkbox-transition-enter(opacity, 0, 80ms),
+                mdc-checkbox-transition-enter(transform, 0, 80ms);
+  }
+
+  .mdc-checkbox:hover .mdc-checkbox__native-control:focus ~ .mdc-checkbox__background::before {
+    opacity: map-get($mdc-ripple-dark-ink-opacities, hover) +
+                 map-get($mdc-ripple-dark-ink-opacities, focus);
+  }
+
+  .mat-ripple-element {
+    opacity: map-get($mdc-ripple-dark-ink-opacities, press);
+  }
+
+  &._mat-animation-noopable {
+    *,
+    *::before {
+      transition: none !important;
+      animation: none !important;
+    }
+  }
+
+  label:empty {
```

Comment:
This is a little fragile, because we could break it just by introducing a comment or a newline inside the template.

Reply:
Yeah it's not perfect... do you have a better suggestion?

Reply:
I think we have something in the checkbox already that uses `cdkObserveContent`. I'm not a fan of that solution either since we have to use a MutationObserver and touch the DOM, but it should be a bit more reliable.

Reply:
I think I'm inclined to just leave this and see if people file issues about it. I don't think this edge case is worth the size and performance cost of `cdkObserveContent`